src/material.h

```diff
@@ -28,12 +30,44 @@ class material {
     __device__ virtual bool scatter(
         const ray& r_in, const hit_record& rec, float3& attenuation, ray& scattered, curandState  *local_rand_state) const{
         return false;
-        }
+    }
     __device__ virtual float3 emitted() const {
         return make_float3(0.0, 0.0, 0.0);
     }
 };
 
+class UniversalMaterial {
+public:
+    UniversalMaterial(float3 baseColorFactor, BaseColorTexture* baseColorTexture) : 
+        baseColorFactor_{baseColorFactor},
+        baseColorTexture_{baseColorTexture} {}
+
+    __device__ bool scatter( 
+        const ray& r_in, 
+        const hit_record& rec, 
+        float3& attenuation, 
+        ray& scattered, 
+        curandState *local_rand_state
+    ) const {
+        float3 scatter_direction = rec.normal + random_in_unit_sphere(local_rand_state);
+        // Catch degenerate scatter direction
+        if (near_zero(scatter_direction)) {
+            scatter_direction = rec.normal;
+        }
+        scattered = ray(rec.p, scatter_direction);
+
+        attenuation = baseColorTexture_->value(rec.texCoord, rec.p);
+        return true;
+    }
+
+    __device__ float3 emitted() const {

```

Comment:
emitted should propably take in a hit_record argument for texture cordinates. Case where objets emits light and has a texture.

Reply:
will be done in the next prs.